src/Microsoft.Health.Encryption/Customer/Health/IDataStoreStateTestProvider.cs

```diff
@@ -0,0 +1,14 @@
+// -------------------------------------------------------------------------------------------------
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License (MIT). See LICENSE in the repo root for license information.
+// -------------------------------------------------------------------------------------------------
+
+using System.Threading;
+using System.Threading.Tasks;
+
+namespace Microsoft.Health.Encryption.Customer.Health;
+
+internal interface IInaccessibleStateTestProvider
+{
+    Task AssertHealthAsync(CancellationToken cancellationToken = default);
```

Comment:
I am generally not a fan of exception-based flow control. One of the reasons being that it is quite slow. Can this interface return something instead?

Reply:
Hm ok - so the reason I threw the exception was because I wanted to exception to be passed down to the CMK Health, which would then get passed to the health checks / health state. 

I updated this to instead just return CustomerKeyHealth, which contains the exception. Also ended up merging these into a single interface since they are doing / returning the same thing